packages/web3-eth-contract/src/contract.ts

```diff
@@ -995,14 +1048,20 @@ export class Contract<Abi extends ContractAbi>
 			return {
 				arguments: abiParams,
 				call: async (options?: NonPayableCallOptions, block?: BlockNumberOrTag) =>
-					this._contractMethodCall(abi, params, errorsAbis, options, block),
+					this._contractMethodCall(methodAbi, params, errorsAbis, options, block),
 				send: (options?: NonPayableTxOptions) =>
-					this._contractMethodSend(abi, params, options), // TODO: refactor to parse errorsAbi
+					this._contractMethodSend(methodAbi, params, options), // TODO: refactor to parse errorsAbi
```

Comment:
Same as the TODO above:
Could you, please, create an issue for this TODO and add its number here?